common/types.go

```diff
@@ -712,17 +712,20 @@ func ConstructError(statusCode int, b []byte) error {
 	var err struct {
 		Error *Error
 	}
-	if e := json.Unmarshal(b, &err); e != nil || err.Error == nil{
+	if e := json.Unmarshal(b, &err); e != nil || err.Error == nil {
 		// return normal error
 		return fmt.Errorf("%d: %s", statusCode, string(b))
 	}
+	err.Error.HTTPReturnedStatusCode = statusCode
 	err.Error.rawData = b
 	return err.Error
 }
 
-// Error is redfish error response object for 400
+// Error is redfish error response object for HTTP status codes different from 200, 201 and 204
 type Error struct {
 	rawData []byte
+	// An integer that represents the status code returned by the API
+	HTTPReturnedStatusCode int
```

Comment:
Seems reasonable. And the JSON marshalling will safely ignore this field since the return data won't include it.